plugins/utils/alpha.nix

```diff
@@ -139,9 +146,24 @@ in
     let
       layoutDefined = cfg.layout != [ ];
       themeDefined = cfg.theme != null;
+
+      opt = options.plugins.alpha;
     in
     mkIf cfg.enable {
-      extraPlugins = [ cfg.package ] ++ (optional cfg.iconsEnabled pkgs.vimPlugins.nvim-web-devicons);
+      # TODO: deprecated 2024-08-29 remove after 24.11
+      warnings = lib.mkIf opt.iconsEnabled.isDefined [
+        ''
+          nixvim (plugins.alpha):
+          The option definition `plugins.alpha.iconsEnabled' in ${showFiles opt.iconsEnabled.files} has been deprecated; please remove it.
+          You should use `plugins.alpha.iconsPackage' instead.
+        ''
+      ];
```

Comment:
Is there a reason this can't just be a `mkRenamedOptionModule`?

Reply:
Would that support different data types? We went from a boolean to a package

Reply:
Ah, one is a nullable package, the other is a Boolean enable option.